src/core/chp.jl

```diff
@@ -162,7 +164,8 @@ function CHP(d::Dict;
         :thermal_efficiency_full_load => chp.thermal_efficiency_full_load,
```

Comment:
Should we assign `thermal_efficiency_full_load` and `_half_load` to zero if `is_electric_only=true`?

Reply:
I agree; this is updated in df24331b80aaeb12af07a8f601a328ead6acfcc8.

In case we get conflicting user inputs (thermal_efficiency_full_load > 0.0 and is_electric_only == true, for example), I set up a warning and change efficiency to zero in e362ae13c40eee29f3ea0799477e3779a4372a8c as well.

Reply:
I was adding a test to make sure the thermal_efficiency_full_load is being set to zero when is_electric_only == true, just under the tests you added, but I'm getting NaN for thermal_efficiency_full_load.

Reply:
My change lets the CHP defaults processing first assign the thermal_efficiency value (non-zero), and then relies on your later checks if >0 then assign to zero. Before, this was setting the value for thermal_efficiency_full_load to 0.0 in custom_chp_inputs, and then when it checked for a value of NaN, it never updated the chp.thermal_efficiency_full_load (because it was 0.0) so it kept it at the default NaN (when not input).